cli/bpmetadata/cmd.go

```diff
@@ -77,8 +77,22 @@ func generate(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	for _, path := range allBpPaths {
-		err = generateMetadataForBpPath(path)
+	for _, modPath := range allBpPaths {
+		// check if module path has readme.md
+		_, err := os.Stat(path.Join(modPath, readmeFileName))
+
+		// throw an error and exit if root module doesn't have a readme.md
+		if !strings.Contains(modPath, modulesPath+"/") && err != nil {
+			return fmt.Errorf("Top-level module does not have a readme. Details: %w\n", err)
+		}
```

Comment:
It seems like if we move this check to after line 59 we can fail fast and potentially simplify the logic?

Reply:
done.